src/main/java/fi/dy/masa/tweakeroo/event/InputHandler.java

```diff
@@ -94,36 +88,35 @@ public boolean onMouseClick(int mouseX, int mouseY, int eventButton, boolean eve
     {
         MinecraftClient mc = MinecraftClient.getInstance();
 
-        if (GuiUtils.getCurrentScreen() == null && mc.player != null && mc.player.isCreative() &&
-            eventButtonState && mc.options.useKey.matchesMouse(eventButton) &&
-            FeatureToggle.TWEAK_ANGEL_BLOCK.getBooleanValue() &&
-            mc.crosshairTarget != null && mc.crosshairTarget.getType() == HitResult.Type.MISS)
+        if (mc.world == null || mc.player == null || mc.interactionManager == null || mc.crosshairTarget == null)
         {
-            BlockPos posFront = PositionUtils.getPositionInfrontOfEntity(mc.player);
+            return false;
+        }
 
-            if (mc.world.isAir(posFront))
-            {
-                Direction facing = PositionUtils.getClosestLookingDirection(mc.player).getOpposite();
-                Vec3d hitVec = PositionUtils.getHitVecCenter(posFront, facing);
-                BlockHitResult context = new BlockHitResult(hitVec, facing, posFront, false);
-                ItemStack stack = mc.player.getMainHandStack();
+        if (GuiUtils.getCurrentScreen() != null)
+        {
+            return false;
+        }
 
-                if (stack.isEmpty() == false && stack.getItem() instanceof BlockItem)
-                {
-                    mc.interactionManager.interactBlock(mc.player, Hand.MAIN_HAND, context);
-                    return true;
-                }
+        if (mc.player.isCreative() && FeatureToggle.TWEAK_ANGEL_BLOCK.getBooleanValue() && eventButtonState && mc.options.useKey.matchesMouse(eventButton) && mc.crosshairTarget.getType() == HitResult.Type.MISS)
+        {
+            Vec3d eyePos = mc.player.getEyePos();
+            Vec3d rotVec = ((IMixinEntity) mc.player).tweakeroo_getRotationVector(mc.player.getPitch(), mc.player.getYaw());
```

Comment:
There is no need to add this `@Invoker`, you can instead use the following method from `Entity`:
```java
public final Vec3d getRotationVec(float tickDelta)
```
So basically just call `getRotationVec(1.0f);`.